rectangle.js

```diff
@@ -1,58 +1,37 @@
-var Rectangle = function(p){
-    var rectangle={};
-    var program = p;
-    var vao;
-    var vaoAttrib;
+class Rectangle {
+    
+    constructor( center, width, height, program ) {
+        this.program = program;
 
-    rectangle.initialize = function(center, width, height){
-        vao = gl.createBuffer();
-        gl.bindBuffer(gl.ARRAY_BUFFER, vao);
-        var vertices = [
+        this.vao = gl.createBuffer();
+        gl.bindBuffer( gl.ARRAY_BUFFER, this.vao );
+        
+        this.vertices = [
             vec2(center[0] + width / 2, center[1] + height / 2),
             vec2(center[0] + width / 2, center[1] - height / 2),
             vec2(center[0] - width / 2, center[1] + height / 2),
             vec2(center[0] - width / 2, center[1] - height / 2)
         ];
-        gl.bufferData(gl.ARRAY_BUFFER, flatten(vertices), gl.STATIC_DRAW);
-        vaoAttrib = gl.getAttribLocation(program, "vPosition");
-        gl.enableVertexAttribArray(vaoAttrib);
-        gl.vertexAttribPointer(vaoAttrib, 2, gl.FLOAT, false, 0, 0);
-
-        gl.bindBuffer(gl.ARRAY_BUFFER, null);
-    }
-
-    rectangle.draw = function(posX){
-        gl.useProgram(program);
         
-        gl.uniform1f(program.x, posX);
-        gl.bindBuffer(gl.ARRAY_BUFFER, vao);
-        gl.enableVertexAttribArray(vaoAttrib);
-        gl.vertexAttribPointer(vaoAttrib, 2, gl.FLOAT, false, 0, 0);
-        console.log(gl.NO_ERROR == gl.getError());
-        gl.drawArrays(gl.TRIANGLE_STRIP, 0, 4);
-    }
-
-//     var indicesArray = [0, 3, 1, 1, 3, 2];
-//     ebo = gl.createBuffer();
-//     gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, ebo);
-//     gl.bufferData(gl.ELEMENT_ARRAY_BUFFER,  new Uint16Array(indicesArray), gl.STATIC_DRAW);
-//     gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, null);
-    
-//     console.log(gl.getError() == gl.NO_ERROR);
-// };
+        gl.bufferData( gl.ARRAY_BUFFER, flatten(this.vertices), gl.STATIC_DRAW );
+        
+        this.vao_attr = gl.getAttribLocation( this.program, 'vPosition' );
+        gl.enableVertexAttribArray( this.vao_attr );
+        gl.vertexAttribPointer( this.vao_attr, 2, gl.FLOAT, false, 0, 0 );
 
-// player.draw = function(program){
-//     gl.useProgram(program);
+        gl.bindBuffer( gl.ARRAY_BUFFER, null );
+    }
 
-//     //gl.uniform1f(program.time, 100);
+    draw( x, y ) {
+        gl.useProgram( this.program );
 
-//     gl.bindBuffer(gl.ARRAY_BUFFER, vBuffer);
-//     gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, ebo);
-//     gl.vertexAttribPointer(vPosition, 2, gl.FLOAT, false, 0, 0);
-//     gl.enableVertexAttribArray(vPosition);
+        gl.uniform1f( this.program.x, x );
+        gl.uniform1f( this.program.y, y );
 
-//     gl.drawElements( gl.TRIANGLES, 6, gl.UNSIGNED_SHORT,0);
-//     console.log(gl.getError() == gl.NO_ERROR);
```

Comment:
Do not remove those commented out lines pls

Reply:
I'm not sure about the validity of commented codes with this new class syntax, but will keep them

Reply:
This is what I am going to use on next session